lib/homekit-device.js

```diff
@@ -2830,18 +2865,21 @@ class HomeKitDevice extends Device {
         };
       });
 
-    this.readPromise = this.client.getCharacteristics(characteristics)
-      .then((result) => {
-        this.handleCharacteristicEvent(result);
-      })
-      .catch((e) => {
-        console.error(`Error reading characteristics for device ${this.deviceID
-        }: ${e}`);
-      }).then(() => {
-        this.readPromise = null;
-        this.pollTimeout = setTimeout(this.readBLECharacteristics.bind(this),
-                                      BLE_POLL_INTERVAL);
-      });
+    this.readPromise = this.adapter.queueBLEOperation(() => {
+      return this.client.getCharacteristics(characteristics)
+        .then((result) => {
+          this.handleCharacteristicEvent(result);
+        })
+        .catch((e) => {
+          console.log(
+            `Error reading characteristics for device ${this.deviceID}: ${e}`
+          );
+        }).then(() => {
```

Comment:
nit: I noticed that the .then and .catch above started on a new line, where this one didn't.

Also, shouldn't it go .then().then().catch() rather than .then().catch().then() ?

Reply:
Nope, I intentionally wanted to catch in that order. This is my form of `.finally()`, since we don't have that in Node v8.